cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/DefaultKafkaQuotasManager.java

```diff
@@ -0,0 +1,157 @@
+ */
+package io.strimzi.operator.cluster.operator.assembly;
+
+import io.strimzi.api.kafka.model.kafka.KafkaResources;
+import io.strimzi.api.kafka.model.kafka.quotas.QuotasPlugin;
+import io.strimzi.api.kafka.model.kafka.quotas.QuotasPluginKafka;
+import io.strimzi.operator.cluster.model.KafkaCluster;
+import io.strimzi.operator.common.AdminClientProvider;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.ReconciliationLogger;
+import io.strimzi.operator.common.auth.PemAuthIdentity;
+import io.strimzi.operator.common.auth.PemTrustSet;
+import io.vertx.core.Future;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.quota.ClientQuotaAlteration;
+import org.apache.kafka.common.quota.ClientQuotaEntity;
+import org.apache.kafka.common.quota.ClientQuotaFilter;
+import org.apache.kafka.common.quota.ClientQuotaFilterComponent;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Class containing methods for handling the configuration around {@link QuotasPlugin}
+ */
+public class DefaultKafkaQuotasManager {
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(DefaultKafkaQuotasManager.class.getName());
+
+    /**
+     * {@link ClientQuotaEntity} for the default users entity
+     * When `null` is set for the ClientQuotaEntity with type USER, the default quotas for all users are used/configured
+     */
+    public static final ClientQuotaEntity DEFAULT_USER_ENTITY = new ClientQuotaEntity(Collections.singletonMap(ClientQuotaEntity.USER, null));
+
+    /**
+     * Returns empty instance of the {@link QuotasPluginKafka}
+     * This is used in case when we need to reset the default Kafka quotas
+     *
+     * @return  empty instance of the {@link QuotasPluginKafka}
+     */
+    private static QuotasPluginKafka emptyQuotasPluginKafka() {
+        QuotasPluginKafka quotasPluginKafka = new QuotasPluginKafka();
+
+        quotasPluginKafka.setConsumerByteRate(null);
+        quotasPluginKafka.setControllerMutationRate(null);
+        quotasPluginKafka.setProducerByteRate(null);
+        quotasPluginKafka.setRequestPercentage(null);
+
+        return quotasPluginKafka;
+    }
+
+    /**
+     * Creates list of {@link ClientQuotaAlteration.Op} based on the configuration in {@link QuotasPluginKafka}.
+     * This list will be then passed to the Admin client and the default user quotas will be set.
+     *
+     * @param quotasPluginKafka     configuration of the quotas
+     *
+     * @return  list of {@link ClientQuotaAlteration.Op}
+     */
+    private static List<ClientQuotaAlteration.Op> getListOfClientQuotaAlterationOps(QuotasPluginKafka quotasPluginKafka) {
```

Comment:
This does not get anything -> maybe you could call it `prepare...` or something like that.